### PR TITLE
chore: Upgrade mantine to 7.4.2

### PR DESCRIPTION
**Pull request by ysael**

### CODE REVIEW

**Review by julienblin — CHANGES_REQUESTED** [review]
Thanks !

A couple of things:
 - this is missing the version update for the `@bonfhir/storybook`
 - `@tabler/icons` should also probably be updated as well
 - as mentioned in my comment here (https://github.com/bonfhir/bonfhir/issues/199#issuecomment-1878849810), I would recommend we update the `peerDependencies` in the `@bonfhir/mantine` package to only reference mantine `^7`, so that not updating the package won't be an issue for apps.

For monorepo version upgrades, I recommend using the helper function at the root of the monorepo to ensure that all packages are aligned with the same version.
In this case, this should work:

```bash
pnpm packages:update '@mantine/*' '@tabler/*'
```

File: packages/mantine/src/r4b/hooks/use-fhir-form.tsx [hunk 1]
```diff
@@ -35,14 +35,15 @@ export function useFhirForm<
   Values = Record<string, unknown>,
   TransformValues extends _TransformValues<Values> = (values: Values) => Values,
 >(
-  args?: UseFormInput<Values, TransformValues>,
+  args?: UseFormInput<Values, any>,
```
Comment (ysael) [inline packages/mantine/src/r4b/hooks/use-fhir-form.tsx]: @julienblin you might have a more clever solution to fix that one 🤔

Comment (julienblin) [reply packages/mantine/src/r4b/hooks/use-fhir-form.tsx]: Yeah it's a problem of transitivity of generic types.
I would advise against altering the `args` though, as it is the right type.

Here is what I suggest:

```typescript
export function useFhirForm<
  Values = Record<string, unknown>,
  TransformValues extends _TransformValues<Values> = (values: Values) => Values,
>(
  args?: UseFormInput<Values, TransformValues>,
): UseFhirFormReturnType<Values, TransformValues> {
  const { transformValues, ...remainingArgs } = args || {};

  const form = useForm<Values, TransformValues>({
    ...remainingArgs,

    transformValues: ((values: Values) => {
      cleanFhirValues(values);
      return transformValues ? transformValues(values) : values;
    }) as any,
  });
  return {
```

**Review by julienblin — CHANGES_REQUESTED** [review]
There are remaining type problems with mantine that needs to be fixed.

Also, please remember to modify the `r5` version, and then run the `pnpm build` and `pnpm check` commands.

Please take the time to run the sample app and storybook to validate that it still works. This PR was clearly not tested as the package hasn't been built :-)

**Comment by julienblin** [comment]
@ysael - I have documented the upgrade process here: https://bonfhir.dev/docs/contributing#maintain-version-consistency-across-package-dependencies (from #207)

When do you think you will be able to complete this? Mantine is already at version 7.4.1.

**Comment by ysael** [comment]
> @ysael - I have documented the upgrade process here: https://bonfhir.dev/docs/contributing#maintain-version-consistency-across-package-dependencies (from #207)
> 
> When do you think you will be able to complete this? Mantine is already at version 7.4.1.

made the changes thanks, but getting some potentially unexpected behaviour as when I ran `pnpm run build` at the root it deleted those files -> https://github.com/bonfhir/bonfhir/pull/200/commits/09ddcae39e8510c9b43357458bc1e14aa052e041

**Comment by ysael** [comment]
build is indeed giving error trying to figure out how to fix that one -> 
image

**Comment by julienblin** [comment]
Build problems should be addressed with this PR: #233

**Pull request closed without merging**